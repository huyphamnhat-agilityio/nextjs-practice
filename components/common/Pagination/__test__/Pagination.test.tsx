import { render, screen } from "@testing-library/react";

// Components
import { Pagination, PaginationProps } from "..";

jest.mock("next/navigation", () => ({
  __esModule: true,
  useRouter() {
    return {
      prefetch: () => null,
    };
  },
  useSearchParams: () => ({
    get: () => {},
  }),
  usePathname: () => ({
    get: () => {},
  }),
}));

describe("Pagination test cases", () => {
  const setup = (props: PaginationProps) => render(<Pagination {...props} />);

  it("should render correctly", () => {
    const { container } = setup({ total: 10, initialPage: 1 });

    expect(container).toMatchSnapshot();
  });
});
