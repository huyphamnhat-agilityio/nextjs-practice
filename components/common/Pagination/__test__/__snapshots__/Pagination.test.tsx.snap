// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination test cases should render correctly 1`] = `
<div>
  <nav
    aria-label="pagination navigation"
    class="p-2.5 overflow-x-scroll scrollbar-hide flex justify-center m-0"
    data-active-page="1"
    data-dots-jump="5"
    data-slot="base"
    data-total="10"
    role="navigation"
  >
    <ul
      class="flex flex-nowrap h-fit max-w-fit relative gap-1 items-center overflow-visible rounded-medium text-white"
      data-slot="wrapper"
    >
      <span
        aria-hidden="true"
        class="absolute flex overflow-visible items-center justify-center origin-center left-0 select-none touch-none pointer-events-none z-20 bg-primary text-primary-foreground data-[moving=true]:transition-transform !data-[moving=true]:duration-300 opacity-0 data-[moving]:opacity-100 min-w-9 w-9 h-9 text-small rounded-medium"
        data-moving="false"
        data-slot="cursor"
        style="transform: translateX(0px) scale(1);"
      >
        1
      </span>
      <li
        aria-current="true"
        aria-label="pagination item 1 active"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-active="true"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        1
      </li>
      <li
        aria-label="pagination item 2"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        2
      </li>
      <li
        aria-label="pagination item 3"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        3
      </li>
      <li
        aria-label="pagination item 4"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        4
      </li>
      <li
        aria-label="pagination item 5"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        5
      </li>
      <li
        aria-label="dots element"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25 group"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="group-hover:hidden group-data-[focus-visible=true]:hidden"
          fill="none"
          height="1em"
          shape-rendering="geometricPrecision"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
        >
          <circle
            cx="12"
            cy="12"
            fill="currentColor"
            r="1"
          />
          <circle
            cx="19"
            cy="12"
            fill="currentColor"
            r="1"
          />
          <circle
            cx="5"
            cy="12"
            fill="currentColor"
            r="1"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="hidden group-hover:block group-data-[focus-visible=true]:block data-[before=true]:rotate-180"
          fill="none"
          focusable="false"
          height="1em"
          role="presentation"
          shape-rendering="geometricPrecision"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M13 17l5-5-5-5"
          />
          <path
            d="M6 17l5-5-5-5"
          />
        </svg>
      </li>
      <li
        aria-label="pagination item 10"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium text-white hover:bg-primary hover:bg-opacity-25"
        data-slot="item"
        role="button"
        tabindex="0"
      >
        10
      </li>
    </ul>
  </nav>
</div>
`;
